core/embed/rust/build.rs

```diff
@@ -133,6 +140,8 @@ fn prepare_bindings() -> bindgen::Builder {
             "-I../../build/unix",
             "-I../../vendor/micropython/ports/unix",
             "-DTREZOR_EMULATOR",
+            "-DFLASH_BIT_ACCESS=1",
+            "-DFLASH_BLOCK_WORDS=1",
```

Comment:
rust (specifically bindgen) needs to know these constants too. We currently don't have universal solution to pass gcc command line macros to rust, so its done manually. Developing some system that would pass all the macros to rust/bindgen automatically would be nice, but its out of scope here

Reply:
@andrewkozlik Regarding this comment, is there anything that still needs to be done?

Reply:
I don't want to hold up the review with a build script cleanup issue, and I am not looking for a universal solution to pass gcc command line macros to rust. What I am not happy with are the facts that:
1. The `FLASH_*` constants are defined multiple times for no apparent reason.
2. The `FLASH_*` constants are handled differently in the emulator build than in the firmware build.

Regarding point 1: For example in site_scons/boards/stm32f4_common.py we could take the `defines` list (or probably just the three items that are added to the `defines` list) and append them to `RUST_INCLUDES` using `";".join("-D{}={}".format(k,v) for k,v in defines)` instead of manually rewriting all the defined values. Another option, although not as extendable, is to define the two values as Python variables `FLASH_BLOCK_WORDS` and `FLASH_BIT_ACCESS` and fill them in like so:
```py
   ...
    FLASH_BLOCK_WORDS = 1
    FLASH_BIT_ACCESS = 1
    defines += [
        ...
        ("FLASH_BLOCK_WORDS", f"{FLASH_BLOCK_WORDS}"),
        ("FLASH_BIT_ACCESS", f"{FLASH_BIT_ACCESS}"),
    ]
    ...
    env.get("ENV")["RUST_INCLUDES"] = (
        ...
        f"-DFLASH_BLOCK_WORDS={FLASH_BLOCK_WORDS};"
        f"-DFLASH_BIT_ACCESS={FLASH_BIT_ACCESS}"
    )
```

Regarding point 2: In the firmware build the constants are passed to the rust build via the environment variable `RUST_INCLUDES` (not a great name, but that's not important). Why can't we do it the same way in the emulator build? Why should they have to be redefined "manually" in a completely different file deep in the build system instead of being pulled from the environment variable in `embed/rust/build.rs`?

If this is more complicated than I am seeing, then let's create a separate issue for the cleanup of these variables, but it seems pretty simple to take care of now. The problem will only continue growing once we add an emulator that uses blockwise storage.

Reply:
While i completely agree with both points, i am against introducing half-measures here to address them only to redo it later properly. 

Neither of the issues originates here, e.g. `STM32_HAL_H` macro is defined exactly the same duplicitous way like new macros, and emulator macros being defined in `build.rs`

Ad 1)
yes this would solve this for these three constants, but to solve this properly you need to address this across all the defines across all boards, and for include paths too and those have different path, so that needs addressing too. 

Ad 2)
yes this would be desirable, but currently there is no `RUST_INCLUDES` in emulator build script, nor the `board` definitions where this originate in hw builds. We should introduce both, but its clearly out of scope here.

The separate issue has already been created (https://github.com/trezor/trezor-firmware/issues/3526) , i would say lets move on.